dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.idl

```diff
@@ -31,13 +96,28 @@ namespace Microsoft.Windows.AppNotifications.Builder
         Critical,
     };
 
+    [contract(AppNotificationBuilderContract, 2)]
+    enum AppNotificationButtonType
+    {
+        Standard,
+        Snooze,
+        Dismiss,
+    };
+
     [contract(AppNotificationBuilderContract, 1)]
     runtimeclass AppNotificationButton
     {
         AppNotificationButton();
         AppNotificationButton(String content);
 
+        [contract(AppNotificationBuilderContract, 2)]
+        static AppNotificationButton MakeSnoozeButton(String inputId);
```

Comment:
Should inputId be optional as in the Dismiss button?

Reply:
It's not optional for dismiss, it simply isn't set (or does nothing if set), while for snooze, it is required. How long is are we going to snooze the dialog if it's not connected to an inputId?